speech/speech_samples.rb

```diff
@@ -12,45 +12,88 @@
 
-  speech_service = Google::Apis::SpeechV1beta1::SpeechService.new
+  speech = Google::Cloud::Speech.new
+  audio  = speech.audio audio_file_path, encoding: :raw, sample_rate: 16000
 
-  speech_service.authorization = Google::Auth.get_application_default(
-    %[ https://www.googleapis.com/auth/cloud-platform ]
-  )
-  # [END initialize_speech_client]
+  results = audio.recognize
+  result  = results.first
 
-  speech_service
+  puts "Transcription: #{result.transcript}"
 end
+# [END speech_sync_recognize]
 
-def transcript_from_audio_file audio_file_path:
-  speech_service = initialize_speech_client
+# [START speech_sync_recognize_gcs]
+def speech_sync_recognize_gcs project_id:, storage_path:
+  # project_id   = "Your Google Cloud project ID"
+  # storage_path = "Path to file in Cloud Storage, eg. gs://bucket/audio.raw"
 
-  # [START transcript_from_audio_file]
-  # audio_file_path = "Path to local audio file"
+  require "google/cloud/speech"
 
-  request        = Google::Apis::SpeechV1beta1::SyncRecognizeRequest.new
-  request.audio  = { content: File.read(audio_file_path) }
-  request.config = { encoding: "LINEAR16", sample_rate: 16000 }
+  speech = Google::Cloud::Speech.new
+  audio  = speech.audio storage_path, encoding: :raw, sample_rate: 16000
 
-  response = speech_service.sync_recognize_speech request
+  results = audio.recognize
+  result  = results.first
 
-  if response.results
-    response.results.each do |recognize_result|
-      recognize_result.alternatives.each do |alternative_hypothesis|
-        puts "Text: #{alternative_hypothesis.transcript}"
-      end
-    end
-  end
-  # [END transcript_from_audio_file]
+  puts "Transcription: #{result.transcript}"
 end
+# [END speech_sync_recognize_gcs]
+
+# [START speech_async_recognize]
+def speech_async_recognize project_id:, audio_file_path:
+  # project_id      = "Your Google Cloud project ID"
+  # audio_file_path = "Path to file on which to perform speech recognition"
+
+  require "google/cloud/speech"
+
+  speech = Google::Cloud::Speech.new
+  audio  = speech.audio audio_file_path, encoding: :raw, sample_rate: 16000
+
+  job = audio.recognize_job
+
+  puts "Job started"
+
+  job.wait_until_done!
+
+  results = job.results
+  result  = results.first
+
+  puts "Transcription: #{result.transcript}"
+end
+# [END speech_async_recognize]
+
+# [START speech_async_recognize_gcs]
+def speech_async_recognize_gcs project_id:, storage_path:
+  # project_id      = "Your Google Cloud project ID"
+  # storage_path = "Path to file in Cloud Storage, eg. gs://bucket/audio.raw"
+
+  require "google/cloud/speech"
+
+  speech = Google::Cloud::Speech.new
+  audio  = speech.audio storage_path, encoding: :raw, sample_rate: 16000
+
+  job = audio.recognize_job
+
+  puts "Job started"
+
+  job.wait_until_done!
+
+  results = job.results
+  result  = results.first
+
+  puts "Transcription: #{result.transcript}"
+end
+# [END speech_async_recognize_gcs]
+
+# Deprecated sample below
```

Comment:
After we publish, I'll send a follow-up PR to delete the old samples below (*currently used on the site*)